src/client/dfuse/dfuse_cont.c

```diff
@@ -155,33 +112,27 @@ dfuse_cont_open(fuse_req_t req, struct dfuse_inode_entry *parent,
 	ie->ie_name[NAME_MAX] = '\0';
 
 	atomic_store_relaxed(&ie->ie_ref, 1);
-	ie->ie_dfs = dfs;
+	ie->ie_dfs = dfc;
 
-	dfs->dfs_ino = atomic_fetch_add_relaxed(&fs_handle->dpi_ino_next, 1);
-	ie->ie_stat.st_ino = dfs->dfs_ino;
-	dfs->dfs_ops = &dfuse_dfs_ops;
+	ie->ie_stat.st_ino = dfc->dfs_ino;
 
 	dfs_obj2id(ie->ie_obj, &ie->ie_oid);
 
 	dfuse_reply_entry(fs_handle, ie, NULL, true, req);
-	D_MUTEX_UNLOCK(&fs_handle->dpi_info->di_lock);
 	return;
-
 close:
-	daos_cont_close(dfs->dfs_coh, NULL);
 	D_FREE(ie);
-err_unlock:
-	D_MUTEX_UNLOCK(&fs_handle->dpi_info->di_lock);
+decref:
+	d_hash_rec_decref(&dfp->dfp_cont_table, &dfc->dfs_entry);
```

Comment:
Is this safe to call when `if (rlink)` above is false? I.e. when we don't find an entry in the hash

Reply:
Yes, the rec_find() above is for looking up the inode, but this reference is for the container handle itself.